growthbook/growthbook
# Surface event webhooks in the UI on the Settings ➡️ Webhooks page

### Features and Changes

Surfaces the Event Webhooks UI on the webhooks page: http://localhost:3000/settings/webhooks

This essentially "launches" the event-based webhooks since it will now be visible in the UI and users will be able to create event-based webhooks.

### Dependencies

- [x] https://github.com/growthbook/growthbook/pull/878

### Testing

- Click "Webhooks" under settings in the sidebar to see.
- Create an event webhook with all currently-available events: `feature.created`, `feature.updated`, `feature.deleted`
-  Create a new feature to test the `feature.created` event
- Delete a feature to test the `feature.deleted` event
- To test `feature.updated`:
  - toggle an environment on/off for a feature
  - put a feature in a project and delete that project. 
  - tag a feature and delete that tag
  - update the description of a feature
  - change the default value of a feature
  - add/remove/change any rules for a feature

### Screenshots

State with event webhooks

![image](https://user-images.githubusercontent.com/113377031/210660399-fe14fd85-4427-43d9-b6a5-c348b6524f81.png)

Empty state

![image](https://user-images.githubusercontent.com/113377031/210660360-a44c9fbf-cb50-4a13-a0f4-5b2ca333a208.png)
